src/snowflake/cli/_plugins/snowpark/commands.py

```diff
@@ -362,22 +369,52 @@ def build(
                 )
 
             if any(temp_deps_dir.path.iterdir()):
-                cli_console.step(f"Creating {project_paths.dependencies.name}")
+                dep_artifact = project_paths.get_dependencies_artefact()
+                cli_console.step(f"Creating {dep_artifact.path.name}")
                 zip_dir(
                     source=temp_deps_dir.path,
-                    dest_zip=project_paths.dependencies,
+                    dest_zip=dep_artifact.post_build_path,
                 )
             else:
                 cli_console.step(f"No external dependencies.")
 
     artifacts = set()
-    for entity in get_snowpark_entities(pd).values():
-        artifacts.update(entity.artifacts)
-
     with cli_console.phase("Preparing artifacts for source code"):
-        for artefact in artifacts:
-            artefact_dto = project_paths.get_artefact_dto(artefact)
-            artefact_dto.build()
+        if FeatureFlag.ENABLE_SNOWPARK_BUNDLE_MAP_BUILD.is_enabled():
+            for entity in get_snowpark_entities(pd).values():
+                for artifact in entity.artifacts:
+                    artifacts.add(project_paths.get_artefact_dto(artifact))
+
+            for artefact in artifacts:
+                bundle_map = BundleMap(
+                    project_root=artefact.project_root,
+                    deploy_root=project_paths.deploy_root,
+                )
+                bundle_map.add(
+                    PathMapping(src=str(artefact.path), dest=(artefact.dest or None))
+                )
+
+                if artefact.path.is_file():
+                    for (absolute_src, absolute_dest) in bundle_map.all_mappings(
+                        absolute=True, expand_directories=False
+                    ):
+                        symlink_or_copy(
```

Comment:
we'll need to apply processors as well, is that planned for a future PR? I can probably help migrate processors to a more general location (and the bundle map too while we're at it), if you want the help. I think there are a few things to fix with processors in the process anyway, this is a good opportunity to make those small, under-the-hood changes.

Reply:
I totally forgot about processors, but I thing it should be added in new PR.